packages/shared-ux/button/exit_full_screen/impl/src/exit_full_screen_button.tsx

```diff
@@ -50,9 +50,10 @@ export const ExitFullScreenButton = ({ onExit = () => {}, toggleChrome = true }:
 
     // cleanup the listener
     return () => {
+      onClick();
```

Comment:
Great digging! I am not a big fan of this solution, but I guess it will do the trick.